internal/framework/framework.go

```diff
@@ -33,14 +34,26 @@ type SearchOptions struct {
 type Metadata struct {
 	FrameworkName      string
 	FrameworkVersion   string
-	Deprecated         bool
+	EOLDate            time.Time
+	RemovalDate        time.Time
 	DockerImage        string
 	GitRelease         string
 	Platforms          []Platform
 	CloudRunnerVersion string
 	BrowserDefaults    map[string]string
 }
 
+func (m *Metadata) IsDeprecated() bool {
+	if time.Now().After(m.EOLDate) && !m.IsFlaggedForRemoval() {
```

Comment:
I find it a bit dangerous to have the logic written this way 🤔 Can trip a dev up in the future. A framework that's no longer supported, yet still has some metadata in our system, is semantically still deprecated.

Reply:
You're right, that check has more sense at the warning level rather than here.
Now it's more clear, and more semantically correct.